src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs

```diff
@@ -209,6 +233,114 @@ private static int CalculatePadding(int width, int componentCount)
         }
 
         /// <summary>
+        /// Looks up color values and builds the image from de-compressed RLE8 data.
+        /// Compresssed RLE8 stream is uncompressed by <see cref="UncompressRle8(int, int)"/>
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="pixels">The <see cref="PixelAccessor{TPixel}"/> to assign the palette to.</param>
+        /// <param name="colors">The <see cref="T:byte[]"/> containing the colors.</param>
+        /// <param name="width">The width of the bitmap.</param>
+        /// <param name="height">The height of the bitmap.</param>
+        /// <param name="inverted">Whether the bitmap is inverted.</param>
+        private void ReadRle8<TPixel>(PixelAccessor<TPixel> pixels, byte[] colors, int width, int height, bool inverted)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            var color = default(TPixel);
+            var rgba = default(Rgba32);
+            byte[] data = this.UncompressRle8(width, height);
```

Comment:
I think this could be better using `Buffer2D<Bgr24>` you would pool the array then and avoid the multiplier.

Reply:
@JimBobSquarePants good point! Don't you mean `Buffer2D<byte>` instead? I'm not familiar with the format, but [line 258](https://github.com/georgexcollins/ImageSharp/blob/b2dea16ab5a2b6e55e170ddec3622d8faecd23be/src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs#L258) suggests that values in `data` does some kind of indexing into the `colors` palette.

@georgexcollins `Buffer<T>` and `Buffer2D<T>` is our centralized memory management helper to overcome GC limitations. Currently it abstracts away pooling, but it will be capable of more magic in the future, so it's important to use it everywhere for large buffers.

Reply:
@antonfirsov Sorry yeah, I meant `Buffer2D`, looking below though at the decompression routing I don't know if it's possible to use it though. Depends on what `cmd[0]` yields. 

@georgexcollins What do you think? You're the only one out of us who knows this part of the format 😄